### PR TITLE
[#4818] update author reordering mng command

### LINKED ISSUE #4818: rewrite author order test
**Description of the bug**
This test fails occasionally. Rewrite it removing 2 assertions:
[https://github.com/hydroshare/hydroshare/blob/4372-communities-and-groups-2.0/hs_c[…]ore/tests/api/native/test_reorder_authors_management_command.py](https://github.com/hydroshare/hydroshare/blob/4372-communities-and-groups-2.0/hs_core/tests/api/native/test_reorder_authors_management_command.py#L180)

[https://github.com/hydroshare/hydroshare/blob/4372-communities-and-groups-2.0/hs_c[…]ore/tests/api/native/test_reorder_authors_management_command.py](https://github.com/hydroshare/hydroshare/blob/4372-communities-and-groups-2.0/hs_core/tests/api/native/test_reorder_authors_management_command.py#L152)

Also: rewrite this management command so that it takes a res ID as a param:
https://github.com/hydroshare/hydroshare/blob/4372-communities-and-groups-2.0/hs_core/management/commands/reorder_authors.py#L24

Steps to reproduce the bug:
http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/5750/testReport/junit/hs_core.tests.api.native.test_reorder_authors_management_command/TestReorderAuthorsCommand/test_command_fixes_triplicate_authors/

**Expected behavior**
Test should not be dependent on django .get() order

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #4818 

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4819](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4819)

File: hs_core/management/commands/reorder_authors.py [hunk 1]
```diff
@@ -6,27 +6,36 @@
...
-                                f"\nExpected: {index}, got: {creator.order}")
-                        creator.order = index
-                        creator.save()
-                        is_dirty = True
-                if is_dirty:
-                    set_dirty_bag_flag(res)
+        if not options['resource_id']:
+            raise CommandError('resource_id argument is required')
+        res_id = options['resource_id']
+        res = BaseResource.objects.filter(short_id=res_id).first()
```
Comment (pkdash) [inline hs_core/management/commands/reorder_authors.py:26]: `res` can be None

**Review by pkdash — CHANGES_REQUESTED** [review]
A small fix needed.

File: hs_core/management/commands/reorder_authors.py (hunk 1 above)
Comment (devincowan) [reply hs_core/management/commands/reorder_authors.py:26]: thanks @pkdash, I added a `raise` for the case of res = None

**Review by pkdash — APPROVED** [review]

**Pull request merged**